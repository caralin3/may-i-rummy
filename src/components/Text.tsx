import React from 'react';
import { Text as RNText, TextStyle } from 'react-native';
import { FontSizes, GutterSizes } from '../appearance';

export interface TextProps {
  align?: TextStyle['textAlign'];
  bold?: boolean;
  gutters?: {
    bottom?: keyof typeof GutterSizes;
    left?: keyof typeof GutterSizes;
    right?: keyof typeof GutterSizes;
    top?: keyof typeof GutterSizes;
  };
  italic?: boolean;
  size?: keyof typeof FontSizes;
  style?: TextStyle;
  // textColor?: keyof typeof defaultTheme.colors;
}

export const Text: React.FC<TextProps> = ({
  align,
  bold,
  children,
  // textColor,
  gutters,
  italic,
  size = 'md',
  style,
}) => (
  <RNText
    style={{
      ...style,
      // color: textColor
      //   ? defaultTheme.colors[textColor].toString()
      //   : !!!!style && !!style.color
      //   ? style.color
      //   : undefined,
      fontSize: FontSizes[size],
      fontStyle: italic ? 'italic' : undefined,
      fontWeight: bold ? 'bold' : undefined,
      lineHeight: 24,
      paddingBottom: !!gutters && !!gutters.bottom ? GutterSizes[gutters.bottom] : undefined,
      paddingLeft: !!gutters && !!gutters.left ? GutterSizes[gutters.left] : undefined,
      paddingRight: !!gutters && !!gutters.right ? GutterSizes[gutters.right] : undefined,
      paddingTop: !!gutters && !!gutters.top ? GutterSizes[gutters.top] : undefined,
      textAlign: align ? align : undefined,
    }}
  >
    {children}
  </RNText>
);
