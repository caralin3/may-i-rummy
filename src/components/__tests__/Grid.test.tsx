import React from 'react';
import renderer from 'react-test-renderer';
import { Col, Row } from '../Grid';
import { Text } from '../Text';

describe('Grid', () => {
  test('renders Row correctly', () => {
    const wrapper = renderer
      .create(
        <Row>
          <Text>Row</Text>
        </Row>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Column correctly', () => {
    const wrapper = renderer
      .create(
        <Row>
          <Col>
            <Text>Col</Text>
          </Col>
          <Col>
            <Text>Col</Text>
          </Col>
        </Row>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
