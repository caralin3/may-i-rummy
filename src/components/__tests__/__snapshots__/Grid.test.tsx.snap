// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders Column correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
        "flex": undefined,
        "justifyContent": "space-between",
        "paddingHorizontal": undefined,
        "paddingVertical": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": undefined,
        "justifyContent": "center",
        "padding": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": undefined,
          "fontSize": 24,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "lineHeight": 24,
          "paddingBottom": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "textAlign": undefined,
        }
      }
    >
      Col
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": undefined,
        "justifyContent": "center",
        "padding": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": undefined,
          "fontSize": 24,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "lineHeight": 24,
          "paddingBottom": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "textAlign": undefined,
        }
      }
    >
      Col
    </Text>
  </View>
</View>
`;

exports[`Grid renders Row correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
        "flex": undefined,
        "justifyContent": "space-between",
        "paddingHorizontal": undefined,
        "paddingVertical": undefined,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": undefined,
        "fontSize": 24,
        "fontStyle": undefined,
        "fontWeight": undefined,
        "lineHeight": 24,
        "paddingBottom": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "textAlign": undefined,
      }
    }
  >
    Row
  </Text>
</View>
`;
